### PR TITLE
Adding SleepPhysionetChallenge2018 dataset

### PR DESCRIPTION
**Pull request by bruAristimunha**
Most of code was from https://github.com/hubertjb/dynamic-spatial-filtering

### CODE REVIEW

**Comment by robintibor** [comment]
Do we really need to add these large csv etc files directly into the codebase? Is there not any other way, like are they hosted somewhere and downloadable? Or is this a common practise to do this way? Seems a bit unexpected to me?

**Comment by bruAristimunha** [comment]
Indeed, I don't think it's a good idea to add this big file here. But, it seems like this is what MNE does in the other sleep dataset. Any idea how to solve this point, @agramfort and @hubertjb?

File: braindecode/datasets/sleep_physio_challe_18.py [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+
+from braindecode.datasets import BaseDataset, BaseConcatDataset
+
+
+PC18_DIR = op.join(op.dirname(__file__), "data", "pc18")
+PC18_RECORDS = op.join(PC18_DIR, "sleep_records.csv")
+PC18_INFO = op.join(PC18_DIR, "age-sex.csv")
+PC18_URL = "https://physionet.org/files/challenge-2018/1.0.0/"
+PC18_SHA1_TRAINING = op.join(PC18_DIR, "training_SHA1SUMS")
+PC18_SHA1_TEST = op.join(PC18_DIR, "test_SHA1SUMS")
```
[the remark this answers is not in the document]
Comment (bruAristimunha) [reply braindecode/datasets/sleep_physio_challe_18.py:35]: There is!

File: braindecode/datasets/sleep_physio_challe_18.py [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+        PC18_INFO: PC18_METAINFO_URL + "age-sex.csv",
+        PC18_SHA1_TRAINING: PC18_METAINFO_URL + "training_SHA1SUMS",
+        PC18_SHA1_TEST: PC18_METAINFO_URL + "test_SHA1SUMS",
+    }
+
+    for file_path, url in files_to_check.items():
+        _download_if_missing(file_path, url)
+
+
+def update_pc18_sleep_records(fname=PC18_RECORDS):
```
Comment (bruAristimunha) [inline braindecode/datasets/sleep_physio_challe_18.py:66]:
```suggestion
def _update_pc18_sleep_records(fname=PC18_RECORDS):
```

File: braindecode/datasets/sleep_physio_challe_18.py [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+                    *params,
+                )
+            else:
+                arousal_fname = None
+            fnames.append([psg_fname, hea_fname, arousal_fname])
+
+    return fnames
+
+
+def convert_wfdb_anns_to_mne_annotations(annots):
```
Comment (bruAristimunha) [inline braindecode/datasets/sleep_physio_challe_18.py:207]:
```suggestion
def _convert_wfdb_anns_to_mne_annotations(annots):
```

File: braindecode/datasets/sleep_physio_challe_18.py [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+            annots = wfdb.rdann(
+                op.splitext(raw_fname[0])[0],
+                "arousal",
+                sampfrom=0,
+                sampto=None,
+                shift_samps=False,
+                return_label_elements=["symbol"],
+                summarize_labels=False,
+            )
+            mne_annots = convert_wfdb_anns_to_mne_annotations(annots)
```
Comment (bruAristimunha) [inline braindecode/datasets/sleep_physio_challe_18.py:356]:
```suggestion
            mne_annots = _convert_wfdb_anns_to_mne_annotations(annots)
```

**Comment by bruAristimunha** [comment]
@gustavohenriquesr,Just to check, you had to do something to make this dataset fully work?

**Comment by gustavohenriquesr** [comment]
Hi @bruAristimunha, 

In fact, curiously I had trouble creating the windows for the dataset (the labels weren't being assigned), but when I used a local copy of the [sleep_physio_challe_18.py](https://github.com/bruAristimunha/braindecode/blob/pc18-dataset/braindecode/datasets/sleep_physio_challe_18.py) file, it worked correctly. 

I didn't get to explore the code further in this regard. I'll check again and if I find anything, I'll let you know. I suggest doing a local test, if you haven't already.

**Comment by bruAristimunha** [comment]
@gustavohenriquesr, It was super silly, test set has no associated label!

**Pull request merged**